### ISSUE TITLE
Add "Open URL and then Perform Auto-type" 

### ISSUE THREAD

**Issue by bernd-wechner** [keepassxreboot/keepassxc#2435]
On the right click menu I find I am always doing a the same sequence of actions. Am I missing something? But I click right-click an entry and click "Open URL" then I press Alt+TAB to get back to KeepassXC then right-click again and click "Perform Auto-Type".

It would be nice to compress this to one click or command. A simple "Open URL and then Perform Auto-Type" command.

If this is already possible, then perhaps this issue is about making it more visible and obvious as I haven't found it and am running through this back and forth use case regularly.

**Comment by droidmonkey** [comment]
It's certainly possible, this would be a good use case for custom macros.

**Comment by fabien4455** [comment]
Is the functionnality has beed added in 2.4.0 ?

**Comment by droidmonkey** [comment]
no

**Comment by fabien4455** [comment]
Ok... :/

**Comment by SohamG** [comment]
I've attempted to add this feature as above, however it only works if the default browser is already open(to any website). If the browser isn't open, auto type triggers before the browser opens the URL and focuses the username field.

Looking for feedback.

CC - @droidmonkey

**Comment by droidmonkey** [comment]
You wont be able to overcome this issue very easily across all platforms. My opinion is that this should not be implemented. We have global auto type shortcut and also browser integration extension. Both of those overcome this timing issue already.

**Comment by SohamG** [comment]
Yea makes sense, it would've been "nice to have" but doesn't seem feasible. Thanks for the reply

**Comment by bernd-wechner** [comment]
I mean I have zero problem with the requirement that the browser be open already. But I wasn't aware of the browser extension. Just checked it out, nice but not really user friendly (yet?). I mean I click "Connected Databases" and none are listed and I see no way of listing them. So I read the github page for the browser extension and it talks about a proxy that's installed with the latest fork. And do I have it/ I have 2.3.1 the latest Ubuntu repo build I guess, so don't know if I have the proxy, but heck I could install it so I check that doesn't come with any easy install options .... it seems to be in development which is great!

In mean time, kudos to SohamG for implementing this and I'm not sure why it shouldn't be implemented. If it relies on the browser being open fine, the browser extensions it seems relies on the KeePassXC client being open too. And if there's a way to wait on the browser to start and open the page before sending keystrokes that would be awesome too.

**Comment by SohamG** [comment]
The issue with the browser being open requirement is if it is not open the functionality would fail without generating any error text or the like which would result in a very unpolished/crude experience. There would be no error text cause it doesn't know if the default browser is open/there is a timing issue. 

Talking about the timing issue, it might be possible to fix this by communicating with the browser extension, which I haven't looked into at all.

The browser integration extension is good though, and I'm sure you wont need the proxy script as long as you aren't using the snap version. The latest version in the [ppa](https://launchpad.net/~phoerious/+archive/ubuntu/keepassxc) is 2.4.1. There's a guide on the [official website](https://keepassxc.org/docs/keepassxc-browser-migration/) for setting up the extension

**Comment by bernd-wechner** [comment]
I'd rather have the feature with that apparently unpolished experience. It's more unpolished expecting two visits and two menu selections to achieve the one action sequence everyone needs. Thanks for the link though showed me I have to enable it in the client. So did that and now it shows up. Guess it would be easier if the browser plugin just said that ;-).

**Comment by mcqj** [comment]
As not all browsers have a browser extension, the open URL, CTRL-TAB, Auto-type is the standard experience on Mac (e.g. with Safari). If there is no work-around, so be it but just wanted to throw in my tuppence that the browser extension is not a solution for everyone

**Comment by droidmonkey** [comment]
We will not be implementing this, just open the URL then press your Auto-Type global hotkey when it is loaded. There are too many timing issues associated with this to be reliable.

**Comment by tantebootsy** [comment]
If anyone is willing to implement this feature for use with Mac and Firefox I'm willing to pay up to 100,- EUR. Sorry I don't have more to give. Maybe some other ppl are interested aswell in paying? Maybe via a hopefully long lasting patch/fork, i don't care. I really miss this feature and I'm annoyed to do the mentioned steps over and over again all day long. I have to deal with a lot of logins every day. Prerequisite: To be used with standard browser. Browser has to be open already.  For me, this would save a lot of nerves and would be the last missing feature to go away from 1Password. They  can do it over there at 1Password so KeepassXC can do it aswell, I guess.
